### DISCUSSION TITLE
[Can't create listen socket] Can't create listen socket: Permission denied at /usr/local/share/perl5/site_perl/Mojo/IOLoop.pm line 124

### DISCUSSION THREAD

**Discussion by heat5** [Difegue/LANraragi#795] [Q&A] [answered]
LANraragi 0.8.9  ,windows11 22621.1413
how to fix this? i try to reinstall or repair it didn't work, but it can sometimes work after the computer reboot
Server PID will be at /home/koyomi/lanraragi/public/temp/server.pid

ｷﾀ━━━━━━(ﾟ∀ﾟ)━━━━━━!!!!!
Enabling AOF on Redis... This might take a while.
[LANraragi] [info] LANraragi 0.8.90 started. (Production Mode)
[LANraragi] [info] Plugin Detected: Chaika.moe
[LANraragi] [info] Plugin Detected: Chaika.moe api.json
[LANraragi] [info] Plugin Detected: ComicInfo
[LANraragi] [info] Plugin Detected: Tag Copier
[LANraragi] [info] Plugin Detected: Add Timestamp Tag
[LANraragi] [info] Plugin Detected: E-Hentai
[LANraragi] [info] Plugin Detected: eze
[LANraragi] [info] Plugin Detected: FAKKU
[LANraragi] [info] Plugin Detected: Hdoujin
[LANraragi] [info] Plugin Detected: Hentag
[LANraragi] [info] Plugin Detected: Hentag Online Lookups
[LANraragi] [info] Plugin Detected: Hitomi
[LANraragi] [info] Plugin Detected: koromo
[LANraragi] [info] Plugin Detected: Mayriad's EH Master Script
[LANraragi] [info] Plugin Detected: Filename Parsing
[LANraragi] [info] Plugin Detected: nHentai
[LANraragi] [info] Script Detected: Blacklist Migration
[LANraragi] [info] Script Detected: Subfolders to Categories
[LANraragi] [info] Script Detected: Source Finder
[LANraragi] [info] Script Detected: nHentai Source Converter
[LANraragi] [info] Downloader Detected: Chaika.moe Downloader
[LANraragi] [info] Downloader Detected: E*Hentai Downloader
Killing process 87 from /home/koyomi/lanraragi/public/temp/minion.pid
Minion will use the Redis database at 127.0.0.1:6379/1
[LANraragi] [info] Successfully connected to Minion database.
[Minion] [info] Starting new Minion worker in subprocess with 12 parallel jobs.
[Minion] [info] Minion worker 137 started
Killing process 93 from /home/koyomi/lanraragi/public/temp/shinobu.pid
[LANraragi] [info] Routing done! Ready to receive requests.
Can't create listen socket: Permission denied at /usr/local/share/perl5/site_perl/Mojo/IOLoop.pm line 124

Error code: Wsl/Service/WSL_E_VM_MODE_INVALID_STATE
Executing the following command on WSL: export LRR_NETWORK=http://*:3000 && export LRR_DATA_DIRECTORY='/mnt/d/Entertainment/CgMangaDoujin/8T' && export LRR_THUMB_DIRECTORY='/mnt/c/Jellyfin and LR DATA/LRdata' && cd /home/koyomi/lanraragi && rm -f public/temp/server.pid && mkdir -p log && mkdir -p content && mkdir -p database && sysctl vm.overcommit_memory=1 && redis-server /home/koyomi/lanraragi/tools/build/docker/redis.conf --dir '/mnt/d/Entertainment/CgMangaDoujin/8T/' --daemonize yes && perl ./script/launcher.pl -f ./script/lanraragi
vm.overcommit_memory = 1
Server PID will be at /home/koyomi/lanraragi/public/temp/server.pid

ｷﾀ━━━━━━(ﾟ∀ﾟ)━━━━━━!!!!!
Can't load application from file "/home/koyomi/lanraragi/script/lanraragi": [KEYS] LOADING Redis is loading the dataset in memory,  at /usr/local/share/perl5/site_perl/Redis.pm line 321.
Compilation failed in require at (eval 88) line 1.

**Comment by Difegue** [comment, marked as answer]
I seem to have missed this question, apologies!  

Looks like your WSL2 setup is broken for some reason -- WSL2 relies on virtualization so I recommend googling that error code to see if you can find any troubleshooting steps there.  
As an alternative, you might be able to switch to WSL1, which doesn't rely on virt.

**Comment by heat5** [comment]
I would like to thankyou for your assistance and advice.  I  still got issues with WSL, so I currently using Docker instead.

I have encountered some new issues. This issue happens in both WSL and Docker. Specifically, when the file names are too long, I face problems with creating covers and reading images. The console displays the following error: "Filename too long at /home/koyomi/lanraragi/script/../lib/LANraragi/Utils/Archive.pm line 303." Below is the complete error code: script/../lib/LANraragi/Utils/Archive.pm line 303.
[2023-09-09 07:10:43.95549] [557] [error] [2C-bpNaiX19j] mkdir /home/koyomi/lanraragi/public/temp/94358480c3393f1db45070f4a4fef011e3da3b64/(C96) [ãã»ãããããããããããâ¥ãã£â¥ãã£â¥ (ããããããã ã)] æ½å
¥å
ã§å«çè¦³ã¼ã­ã®ç¯ç½ªè
ã«å¤æ
ã¢ãã«èå¥ä»ããããµå¯ããå
¬è¡ä¾¿å¨å¨¼å©¦ã«ãªã£ã¡ãã£ã­£ç¾©ã®Yè±ã¡ããæ¬ (å¯¾é­å¿ã¦ã­ã«ã¼) [ç
                                                               ¡éªæ°æ¼¢åçµ
                                                                          ] [compr] [PNG8]/: Filename too long at /home/koyomi/lanraragi/script/../lib/LANraragi/Utils/Archive.pm line 303.
________________________________
发件人: Difegue ***@***.***>
发送时间: 2023年5月9日 20:06
收件人: Difegue/LANraragi ***@***.***>
抄送: heat5 ***@***.***>; Author ***@***.***>
主题: Re: [Difegue/LANraragi] [Can't create listen socket] Can't create listen socket: Permission denied at /usr/local/share/perl5/site_perl/Mojo/IOLoop.pm line 124 (Discussion #795)


I seem to have missed this question, apologies!

Looks like your WSL2 setup is broken for some reason -- WSL2 relies on virtualization so I recommend googling that error code to see if you can find any troubleshooting steps there.
As an alternative, you might be able to switch to WSL1, which doesn't rely on virt.

**Reply by Difegue** [reply]
Right -- there's no path length check for temporary/extracted files at the moment, so this is something that can sadly happen with archives that have a folder built-in with a long name.  
It doesn't help that it looks like the unicode parsed incorrectly here, leading to a freakishly long folder name...

This is a Windows-exclusive problem, so you might have some luck enabling long path support: https://www.howtogeek.com/266621/how-to-make-windows-10-accept-file-paths-over-260-characters/

**Comment by heat5** [comment]
I forgot to tell you that I have already enabled Synology eCryptFS Compatibility Mode, but it didn't work .
________________________________
发件人: H x ***@***.***>
发送时间: 2023年9月9日 7:38
收件人: Difegue/LANraragi ***@***.***>
主题: 回复: [Difegue/LANraragi] [Can't create listen socket] Can't create listen socket: Permission denied at /usr/local/share/perl5/site_perl/Mojo/IOLoop.pm line 124 (Discussion #795)

I would like to thankyou for your assistance and advice.  I  still got issues with WSL, so I currently using Docker instead.

I have encountered some new issues. This issue happens in both WSL and Docker. Specifically, when the file names are too long, I face problems with creating covers and reading images. The console displays the following error: "Filename too long at /home/koyomi/lanraragi/script/../lib/LANraragi/Utils/Archive.pm line 303." Below is the complete error code: script/../lib/LANraragi/Utils/Archive.pm line 303.
[2023-09-09 07:10:43.95549] [557] [error] [2C-bpNaiX19j] mkdir /home/koyomi/lanraragi/public/temp/94358480c3393f1db45070f4a4fef011e3da3b64/(C96) [ãã»ãããããããããããâ¥ãã£â¥ãã£â¥ (ããããããã ã)] æ½å
¥å
ã§å«çè¦³ã¼ã­ã®ç¯ç½ªè
ã«å¤æ
ã¢ãã«èå¥ä»ããããµå¯ããå
¬è¡ä¾¿å¨å¨¼å©¦ã«ãªã£ã¡ãã£ã­£ç¾©ã®Yè±ã¡ããæ¬ (å¯¾é­å¿ã¦ã­ã«ã¼) [ç
                                                               ¡éªæ°æ¼¢åçµ
                                                                          ] [compr] [PNG8]/: Filename too long at /home/koyomi/lanraragi/script/../lib/LANraragi/Utils/Archive.pm line 303.
________________________________
发件人: Difegue ***@***.***>
发送时间: 2023年5月9日 20:06
收件人: Difegue/LANraragi ***@***.***>
抄送: heat5 ***@***.***>; Author ***@***.***>
主题: Re: [Difegue/LANraragi] [Can't create listen socket] Can't create listen socket: Permission denied at /usr/local/share/perl5/site_perl/Mojo/IOLoop.pm line 124 (Discussion #795)


I seem to have missed this question, apologies!

Looks like your WSL2 setup is broken for some reason -- WSL2 relies on virtualization so I recommend googling that error code to see if you can find any troubleshooting steps there.
As an alternative, you might be able to switch to WSL1, which doesn't rely on virt.

**Reply by Difegue** [reply]
The eCryptFS compatibility mode only affects file uploads, so sadly it won't help here!

**Comment by heat5** [comment]
Thank you for your reply. I tried the solution you provided in the link, but it didn't work. The console still shows the same error. This is the file where I encountered the problem. Can you please take a look at it?  [ https://pixeldrain.com/u/X52FSzM5 ] or [ https://gofile.io/d/wliPm6 ]


________________________________
发件人: Difegue ***@***.***>
发送时间: 2023年9月9日 19:40
收件人: Difegue/LANraragi ***@***.***>
抄送: heat5 ***@***.***>; Author ***@***.***>
主题: Re: [Difegue/LANraragi] [Can't create listen socket] Can't create listen socket: Permission denied at /usr/local/share/perl5/site_perl/Mojo/IOLoop.pm line 124 (Discussion #795)


Right -- there's no path length check for temporary/extracted files at the moment, so this is something that can sadly happen with archives that have a folder built-in with a long name.
It doesn't help that it looks like the unicode parsed incorrectly here, leading to a freakishly long folder name...

This is a Windows-exclusive problem, so you might have some luck enabling long path support: https://www.howtogeek.com/266621/how-to-make-windows-10-accept-file-paths-over-260-characters/

―
